src/main/java/com/comphenix/protocol/reflect/compiler/StructureCompiler.java

```diff
@@ -289,36 +297,54 @@ private String getCompiledName(StructureModifier<?> source) {
 
 		byte[] data = cw.toByteArray();
 
-		// Call the define method
-		try {
-			if (defineMethod == null) {
-				Method defined = ClassLoader.class.getDeclaredMethod("defineClass",
-					new Class<?>[] { String.class, byte[].class, int.class, int.class });
+		Class<?> clazz = defineClass(data);
+		// DEBUG CODE: Print the content of the generated class.
+		//org.objectweb.asm.ClassReader cr = new org.objectweb.asm.ClassReader(data);
+		//cr.accept(new ASMifierClassVisitor(new PrintWriter(System.out)), 0);
+		return clazz;
+	}
 
-				// Awesome. Now, create and return it.
-				defined.setAccessible(true);
-				defineMethod = defined;
-			}
+	private Class<?> defineClassLegacy(byte[] data) throws InvocationTargetException, IllegalAccessException,
+			NoSuchMethodException {
+		if (defineMethod == null) {
+			Method defined = ClassLoader.class.getDeclaredMethod("defineClass",
+					new Class<?>[]{String.class, byte[].class, int.class, int.class});
 
-			@SuppressWarnings("rawtypes")
-			Class clazz = (Class) defineMethod.invoke(loader, null, data, 0, data.length);
+			// Awesome. Now, create and return it.
+			defined.setAccessible(true);
+			defineMethod = defined;
+		}
+		return (Class<?>) defineMethod.invoke(loader, null, data, 0, data.length);
+	}
+
+	private Class<?> defineClassModern(byte[] data) throws InvocationTargetException, IllegalAccessException,
+			ClassNotFoundException, NoSuchMethodException {
+		if (defineMethod == null || lookup == null)
+		{
+			Class<?> lookupClass = Class.forName("java.lang.invoke.MethodHandles$Lookup");
```

Comment:
Might not be a bad idea to cache these Class#forName calls

Reply:
Yep, definitely. Will add that.